### PR TITLE
EVG-2474 Fix cloud tests

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: cloud/new_ec2_client.go [hunk 1]
```diff
@@ -94,6 +102,12 @@ func (c *awsClientImpl) RunInstances(input *ec2.RunInstancesInput) (*ec2.Reserva
 		func() (bool, error) {
 			output, err = c.EC2.RunInstances(input)
 			if err != nil {
+				if ec2err, ok := err.(awserr.Error); ok {
```
Comment (john-m-liu) [inline cloud/new_ec2_client.go]: should these errors be checking ec2err != nil rather than ok?

File: cloud/new_ec2_client.go [hunk 2]
```diff
@@ -391,5 +453,8 @@ func (c *awsClientMock) GetInstanceInfo(id string) (*ec2.Instance, error) {
 	instance.Placement.AvailabilityZone = makeStringPtr("us-east-1a")
 	instance.InstanceType = makeStringPtr("m3.4xlarge")
 	instance.LaunchTime = makeTimePtr(time.Now())
+	instance.PublicDnsName = makeStringPtr("public_dns_name")
```
Comment (john-m-liu) [inline cloud/new_ec2_client.go]: is this just a placeholder? how come it's needed?

File: cloud/new_ec2_client.go (hunk 1 above)
Comment (bsamek) [reply cloud/new_ec2_client.go]: The error is not nil from the line above. ok verifies that the type assertion was successful. From the docs: "Returns awserr.Error for service API and SDK errors. Use runtime type assertions with awserr.Error's Code and Message methods to get detailed information about the error." (https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/)

File: cloud/new_ec2_client.go (hunk 2 above)
Comment (bsamek) [reply cloud/new_ec2_client.go]: It's returned from the mock when testing this function https://github.com/bsamek/evergreen/blob/ac343b37e85b9634d66448a103184db34450e16e/cloud/new_ec2.go#L534.

**Pull request closed without merging**